### ISSUE TITLE
Consider changing license to GPL

### ISSUE THREAD

**Issue by rpwoodbu** [winfsp/hubfs#11]
In [this commit](https://github.com/winfsp/winfsp/commit/c7780cf7faf14fdcafa5d9d7d0389e13407d6b07), [WinFsp](https://github.com/winfsp/winfsp) was changed from AGPL to GPL. I wonder if the same could be done for HUBFS. Many companies outright forbid the use of AGPL software in any capacity.

I think this is a very promising project, and I would like to contribute to it, but we cannot proceed with it under the AGPL.

**Comment by billziss-gh** [comment]
I would consider changing the license from AGPL to GPL.

Can you please let me know what kind of changes you would like to contribute?

**Comment by rpwoodbu** [comment]
Thanks for considering it!

I have observed stability issues with the cache expiration logic, and I would like to help address that. I would also like to make that more configurable, as it is too aggressive for my use-case. In particular, I don't leave files/directories open in-between builds, so the 30s expiration causes full cache loss between builds.

Another simple feature I'd like to add is a read-only option. We have a large monorepo, and what we want to do is use HubFS as a look-aside read-only view of the full repo, while doing development within an ordinary git worktree using sparse checkouts. We're using Bazel, and it has the ability (via `--package_path`) to look elsewhere for any files that are missing from the main workspace. I've shown that this works (e.g., I can build [Envoy](https://github.com/envoyproxy/envoy) with only the root directory checked-out), but it is important for HubFS to guarantee a pristine view of the repo.

I've only been able to experiment with HubFS on my personal machine with public repos so far, as my company disallows any use of AGPL software, so I'm very glad to hear you'd consider changing the license.
